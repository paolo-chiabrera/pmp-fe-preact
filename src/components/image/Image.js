import { h } from 'preact';

export default function Image({ image }) {
	return (
		<div class="image-details">{image.filename} - {image.source}</div>
	);
}