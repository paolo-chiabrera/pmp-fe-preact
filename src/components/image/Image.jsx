import { h } from 'preact';

export default function Image({ image }) {
  const { filename, meta: { ext } } = image;

  return (
    <div class="image-wrapper">
      <img
        src={`http://images.picmeplease.eu/images/${filename}.${ext}`}
        alt={filename}
      />
    </div>
  );
}
