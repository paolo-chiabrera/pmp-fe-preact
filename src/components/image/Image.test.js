import { h } from 'preact';

import render from 'preact-render-to-string';

import Image from './Image.jsx';

const IMAGE = {
  filename: 'fake',
  meta: {
    ext: 'jpg'
  }
};

describe('Given a Image component', () => {
  it('should render correctly', () => {
    const tree = render(<Image image={IMAGE} />);

    expect(tree).toMatchSnapshot();
  });
});
