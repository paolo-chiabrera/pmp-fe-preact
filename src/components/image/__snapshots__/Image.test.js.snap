// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Image component should render correctly 1`] = `"<div class=\\"image-wrapper\\"><img src=\\"http://images.picmeplease.eu/images/fake.jpg\\" alt=\\"fake\\" /></div>"`;
